sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py

```diff
@@ -2800,18 +2809,84 @@ def register_both_output():
+        pipeline_job = assert_job_cancel(pipeline, client)
 
-        def check_name_version_and_register_succeed(output, output_name, output_version):
-            assert output.name == output_name
-            assert output.version == output_version
-            assert client.data.get(name=output_name, version=output_version)
+        check_name_and_version(pipeline_job.outputs.pipeine_a_output, "p1_output", "v1")
+        check_name_and_version(pipeline_job.jobs["node_2"].outputs.component_out_path, "n2_output", "v1")
+        assert pipeline_job.jobs["node_3"].outputs.component_out_path.name == "n3_output"
+        check_name_and_version(pipeline_job.jobs["sub_node"].outputs.sub_pipeine_a_output, "sub_pipeline", "v1")
 
-        check_name_version_and_register_succeed(pipeline_job.outputs.pipeine_a_output, "p1_output", "v1")
-        check_name_version_and_register_succeed(
-            pipeline_job.jobs["node_2"].outputs.component_out_path, "n2_output", "v1"
-        )
-        check_name_version_and_register_succeed(
-            pipeline_job.jobs["sub_node"].outputs.sub_pipeine_a_output, "sub_pipeline", "v1"
-        )
+    @pytest.mark.skip("https://msdata.visualstudio.com/Vienna/_workitems/edit/2245690")
+    @pytest.mark.disable_mock_code_hash
+    def test_register_output_for_pipeline_component(self, client: MLClient):
+        component = load_component(source="./tests/test_configs/components/helloworld_component.yml")
+        component_input = Input(type="uri_file", path="https://dprepdata.blob.core.windows.net/demo/Titanic.csv")
+
+        @dsl.pipeline()
+        def sub_pipeline():
+            node_1 = component(component_in_path=component_input)  # test use Output to initialize subgraph.jobs.output
+            node_1.outputs.component_out_path = Output(name="sub_pipeline_1_output", version="v1")
+
+            node_2 = component(component_in_path=component_input)  # test we can pass NodeOutput in PipelineComponent
+            node_2.outputs.component_out_path.name = "sub_pipeline_2_output"
+            node_2.outputs.component_out_path.version = "v2"
+
+            return {
+                "sub_node_1": node_1.outputs.component_out_path,
+            }
+
+        @dsl.pipeline()
+        def register_both_output():
+            subgraph = sub_pipeline()
+
+        pipeline = register_both_output()
+        pipeline.settings.default_compute = "cpu-cluster"
+        pipeline_job = assert_job_cancel(pipeline, client)
+
+        check_name_and_version(pipeline_job.jobs["subgraph"].outputs["sub_node_1"], "sub_pipeline_1_output", "v1")
+
+        subgraph_id = pipeline_job.jobs["subgraph"].component
+        subgraph_id = subgraph_id.split(":")
+        subgraph = client.components.get(name=subgraph_id[0], version=subgraph_id[1])
+        check_name_and_version(subgraph.jobs["node_2"].outputs["component_out_path"], "sub_pipeline_2_output", "v2")
+
+    @pytest.mark.disable_mock_code_hash
```

Comment:
Add some comment for this?

Reply:
Please add comment, in case some breaking change for reuse, we need people to aware the root cause